### PR TITLE
feat(j-s): Update Litigants at Court

### PR DESCRIPTION
**Pull request by gudjong**
# Update Litigants at Court

[Skila gögnum um varnaraðila við móttöku máls í Auði](https://app.asana.com/0/1199153462262248/1203337868753454/f)
[Uppfæra gögn um verjanda í Auði fyrir fyrirtöku](https://app.asana.com/0/1199153462262248/1203170952250137/f)
[Skila gögnum um sóknaraðila og sækjanda við móttöku allra mála í Auði](https://app.asana.com/0/1199153462262248/1203170952250139/f)

## What

- Updates court case with prosecutor and defendant info in the court system.
- Sends email to court if an update fails.

## Why

- Reduces manual data entry.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/backend/infra/judicial-system-backend.ts [hunk 1]
```diff
@@ -40,8 +40,8 @@ export const serviceSetup = (): ServiceBuilder<'judicial-system-backend'> =>
       SQS_REGION: 'eu-west-1',
       BLOCKED_API_INTEGRATION: {
         dev: '',
-        staging: 'COURT,POLICE_DOCUMENT,POLICE_CASE',
-        prod: 'POLICE_CASE',
+        staging: 'COURT,COURT_LITIGANTS,POLICE_DOCUMENT,POLICE_CASE',
+        prod: 'COURT_LITIGANT,POLICE_CASE',
```
Comment (brynjarorng) [inline apps/judicial-system/backend/infra/judicial-system-backend.ts:44]: I see there is a difference in the names of the integrations, S missing in the prod statement compared to staging. Is this intentional?

**Review by brynjarorng — APPROVED** [review]
Added one comment to a possible minor issue, otherwise LGTM(Devops)

**Review by Arnarkari93 — CHANGES_REQUESTED** [review]
Great work. I have some conserns about `{ ... } as CaseMessage`. I feel like there should be a better way making this more typesafe without us needing to step in and overwrite the types. I do not have enough experience with typescript to come up with a solution right now.

Other than  that I just have a few nitpicks and also we will need to update `COURT_LITIGANTS` to `COURT_LITIGANT` without the (s)

File: apps/judicial-system/backend/infra/judicial-system-backend.ts (hunk 1 above)
Comment (gudjong) [reply apps/judicial-system/backend/infra/judicial-system-backend.ts:44]: Not intentional, will fix.

Comment (gudjong) [reply apps/judicial-system/backend/infra/judicial-system-backend.ts]: Well spotted

File: apps/judicial-system/backend/migrations/20221215110041-update-institution.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+              { transaction },
+            ),
+            queryInterface.bulkInsert(
+              'institution',
+              [
+                {
+                  id: '0be621ec-c063-4df3-ab15-61f6e421ed7c',
+                  name: 'Lögreglustjórinn á Norðurlandi vestra',
+                  type: 'PROSECUTORS_OFFICE',
+                  active: false,
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/backend/migrations/20221215110041-update-institution.js:103]: yes, this is intentional

File: apps/judicial-system/backend/migrations/20221215110041-update-institution.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+                  default_court_id: '7299ab8f-2fcc-40be-8194-8c2f749b4791',
+                  // eslint-disable-next-line local-rules/disallow-kennitalas
+                  national_id: '6708140330',
+                },
+              ],
+              { transaction },
+            ),
+            queryInterface.bulkInsert(
+              'institution',
+              [
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/backend/migrations/20221215110041-update-institution.js]: I agree with the use of "as". Seems like I struggle with this a lot. I will take a look and see if I can make this work without it.

File: apps/judicial-system/backend/src/app/modules/case/case.service.ts [hunk 4]
```diff
@@ -175,69 +169,126 @@ export class CaseService {
...
+      } as CaseMessage,
+    ]
+  }
+
+  private addMessagesForIndictmentCourtCaseConnectionToQueue(
+    theCase: Case,
+    user: TUser,
+  ): Promise<void> {
+    return this.messageService.sendMessagesToQueue(
+      this.getDeliverProsecutorToCourtMessages(theCase, user).concat(
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/case/case.service.ts]: Done

File: apps/judicial-system/backend/migrations/20221215110041-update-institution.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/backend/migrations/20221215110041-update-institution.js]: For this particular case I simply put the return value into a variable and that fixed the type inference. Strange, but works.

**Review by Arnarkari93 — APPROVED** [review]

**Pull request merged**